pkg/reconciler/taskrun/taskrun.go

```diff
@@ -397,28 +399,21 @@ func (c *Reconciler) handlePodCreationError(tr *v1alpha1.TaskRun, err error) {
 	c.Logger.Errorf("Failed to create build pod for task %q: %v", tr.Name, err)
 }
 
-func updateTaskRunResourceResult(taskRun *v1alpha1.TaskRun, pod *corev1.Pod, logger *zap.SugaredLogger) {
+func updateTaskRunResourceResult(taskRun *v1alpha1.TaskRun, podStatus corev1.PodStatus) error {
 	if taskRun.IsSuccessful() {
-		for _, cs := range pod.Status.ContainerStatuses {
+		for _, cs := range podStatus.ContainerStatuses {
 			if cs.State.Terminated != nil {
 				msg := cs.State.Terminated.Message
-				if msg != "" {
-					if err := updateTaskRunStatusWithResourceResult(taskRun, []byte(msg)); err != nil {
-						logger.Infof("No resource result from %s for %s/%s: %s", cs.Name, taskRun.Name, taskRun.Namespace, err)
-					}
+				r, err := termination.ParseMessage(msg)
+				if err != nil {
+					return err
```

Comment:
i'm curious why in this case and in the `return err` above you didn't want to add some more context to the error with fmt.errorf or similiar? i find i helps in debugging to add a bit of context along the way

Reply:
Done.